File: src/telegram/types/chat-permissions.ts
import * as type from '../types'
import Type from './type'

export declare type JsonChatPermissions = {
	can_send_messages?: type.Boolean
	can_send_media_messages?: type.Boolean
	can_send_polls?: type.Boolean
	can_send_other_messages?: type.Boolean
	can_add_web_page_previews?: type.Boolean
	can_change_info?: type.Boolean
	can_invite_users?: type.Boolean
	can_pin_messages?: type.Boolean
}

export default class ChatPermissions extends Type {
	can_send_messages?: type.Boolean
	can_send_media_messages?: type.Boolean
	can_send_polls?: type.Boolean
	can_send_other_messages?: type.Boolean
	can_add_web_page_previews?: type.Boolean
	can_change_info?: type.Boolean
	can_invite_users?: type.Boolean
	can_pin_messages?: type.Boolean

	defaultObject!: JsonChatPermissions

	constructor(chatPermissions: JsonChatPermissions) {
		super()
		this.defaultObject = chatPermissions

		this.can_send_messages = chatPermissions.can_send_messages
		this.can_send_media_messages = chatPermissions.can_send_media_messages
		this.can_send_polls = chatPermissions.can_send_polls
		this.can_send_other_messages = chatPermissions.can_send_other_messages
		this.can_add_web_page_previews = chatPermissions.can_add_web_page_previews
		this.can_change_info = chatPermissions.can_change_info
		this.can_invite_users = chatPermissions.can_invite_users
		this.can_pin_messages = chatPermissions.can_pin_messages
	}
}

File: telegram-api-json/pygen/ChatPermissions.py

from telelib.bot import DefaultType

# @url https://core.telegram.org/bots/api#chatpermissions# ['Describes actions that a non-administrator user is allowed to take in a chat.']
class ChatPermissions(DefaultType):

    
	# @var can_send_messages Optional. True, if the user is allowed to send text messages, contacts, locations and venues
	@property
	def can_send_messages(self) -> Boolean:
		return self._d["can_send_messages"]
	# @var can_send_media_messages Optional. True, if the user is allowed to send audios, documents, photos, videos, video notes and voice notes, implies can_send_messages
	@property
	def can_send_media_messages(self) -> Boolean:
		return self._d["can_send_media_messages"]
	# @var can_send_polls Optional. True, if the user is allowed to send polls, implies can_send_messages
	@property
	def can_send_polls(self) -> Boolean:
		return self._d["can_send_polls"]
	# @var can_send_other_messages Optional. True, if the user is allowed to send animations, games, stickers and use inline bots, implies can_send_media_messages
	@property
	def can_send_other_messages(self) -> Boolean:
		return self._d["can_send_other_messages"]
	# @var can_add_web_page_previews Optional. True, if the user is allowed to add web page previews to their messages, implies can_send_media_messages
	@property
	def can_add_web_page_previews(self) -> Boolean:
		return self._d["can_add_web_page_previews"]
	# @var can_change_info Optional. True, if the user is allowed to change the chat title, photo and other settings. Ignored in public supergroups
	@property
	def can_change_info(self) -> Boolean:
		return self._d["can_change_info"]
	# @var can_invite_users Optional. True, if the user is allowed to invite new users to the chat
	@property
	def can_invite_users(self) -> Boolean:
		return self._d["can_invite_users"]
	# @var can_pin_messages Optional. True, if the user is allowed to pin messages. Ignored in public supergroups
	@property
	def can_pin_messages(self) -> Boolean:
		return self._d["can_pin_messages"]

